### PR TITLE
Bach: Generate unique table names

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Created `unique_table_test_name` fixture that generates unique table names based on test run UUID, this way we don't have conflicts between GA checks when running against multiple python versions

### CODE REVIEW

File: bach/tests/conftest.py [hunk 1]
```diff
@@ -76,6 +79,35 @@ def pg_engine() -> Engine:
...
+    # sqlalchemy will raise an error if we exceed this length
+    _MAX_LENGTH_TABLE_NAME = 63
+
+    # testrun_uid is an UUID
+    _TEST_RUN_ID_LENGTH = 32
+
+    current_test_name = request.node.originalname[:_MAX_LENGTH_TABLE_NAME - _TEST_RUN_ID_LENGTH - 1]
+    table_name = f'{current_test_name}_{testrun_uid}'
+
+    _RECORDED_TEST_TABLES_PER_ENGINE[engine].append(table_name)
```
Comment (KathiaBarahona) [inline bach/tests/conftest.py:107]: Tried using[ `record_property`](https://docs.pytest.org/en/6.2.x/_modules/_pytest/junitxml.html#record_property) but didn't work :sleepy:

File: bach/tests/functional/bach/test_df_from_pandas.py [hunk 2]
```diff
@@ -143,39 +140,38 @@ def test_from_pandas_non_happy_path(pg_engine):
...
             materialization='table',
         )
         DataFrame.from_pandas(
             engine=pg_engine,
             df=pdf,
             convert_objects=True,
-            name='test_from_pd_table',
+            name=unique_table_test_name,
             materialization='table',
         )
```
Comment (KathiaBarahona) [inline bach/tests/functional/bach/test_df_from_pandas.py:163]: maaaybe can remove this...?

Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_df_from_pandas.py:165]: updated the comment in https://github.com/objectiv/objectiv-analytics/pull/1045/commits/347b73fc663a248855a405f153caacaec75269e4

Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_df_from_pandas.py:163]: neeh reverted.

File: bach/tests/conftest.py [hunk 3]
```diff
@@ -38,10 +38,12 @@
```
Comment (thijs-obj) [inline bach/tests/conftest.py:38]:
```suggestion
The fixture `unique_table_test_name` gives out a unique table name to each test that use it. These tables are
automatically deleted at the end of the test session.
```

**Review by thijs-obj — APPROVED** [review]

File: bach/tests/conftest.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/tests/conftest.py:38]: Done here: https://github.com/objectiv/objectiv-analytics/pull/1045/commits/b151f7e6ee035a009788467fd1ec6dbec76656a8

**Review by jansentom — APPROVED** [review]
Really nice!

**Comment by KathiaBarahona** [comment]
> Didn't remove the xdist groups from this module https://github.com/objectiv/objectiv-analytics/blob/main/bach/tests/functional/bach/test_savepoints.py because I was not really sure how those ones work :/

Managed here https://github.com/objectiv/objectiv-analytics/pull/1045/commits/8baad18a56115d86124af2368b8c64e128b8bdb7

**Pull request merged**